MAVProxy/mavproxy.py

```diff
@@ -548,7 +548,7 @@ def process_mavlink(slave):
     except socket.error:
         return
     try:
-        if slave.first_byte and opts.auto_protocol:
+        if slave.first_byte and (opts.mavversion == "" or opts.auto_protocol):
```

Comment:
I think by this point we should be able to check whether the version has been set in the environment.  `opts` is only in scope because Python sucks.

Yeah, you're only doing what the code currently does in terms of scope, but if the thing is going to be thoroughly tsted it should be Made Right, I think.